Docs: Document alertmanager shuffle sharding (#2220)
grafana/mimir: docs/sources/operators-guide/configuring/configuring-shuffle-sharding/index.md

You can override the compactor shard size on a per-tenant basis setting by `compactor_tenant_shard_size` in the overrides section of the runtime configuration.

### Alertmanager shuffle sharding

> Alertmanager only distributes across replicas by tenant, the state and workload it is not divided any further. The replication factor setting `-alertmanager.sharding-ring.replication-factor` dictates how many replicas are used for a tenant.

Reviewer: ```suggestion
Alertmanager only performs distribution across replicas per tenant. The state and workload is not divided any further. The replication factor setting `-alertmanager.sharding-ring.replication-factor` determines how many replicas are used for a tenant.
```

Reviewer: Please vet for technical accuracy because the first sentence seems like it initially contained a typo.

Reviewer: The verb "to distribute" requires a direct object.

Author: You're right, there was an extra "it".

> As a consequence, shuffle sharding is effectively always enabled for Alertmanager.

Reviewer: ```suggestion
As a result, shuffle sharding is effectively always enabled for Alertmanager.
```
What does "effectively" mean here? Is it or is it not enabled?

Author: It means that there is no explicit method to enable "shuffle sharding", because the effect of limiting tenants to a fixed number of replicas is inherent in the design.
